DigitalLearningSolutions.Web/Controllers/MyAccountController.cs

```diff
@@ -49,23 +56,87 @@ public IActionResult EditDetails()
         }
 
         [HttpPost]
-        public IActionResult EditDetails(EditDetailsViewModel model)
+        public IActionResult EditDetails(EditDetailsViewModel model, string action)
+        {
+            switch (action)

```

Comment:
My IDE recommends changing this to a switch expression, which I think reads much nicer :)